code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -233,18 +235,62 @@
 	bonus_tame_chance = 0
 	pet_bonus = TRUE
 	pet_bonus_emote = "bloops happily!"
+	/// Keeping track of the nuke disk for the functionality of storing it.
+	var/obj/item/disk/nuclear/disky
 
-/mob/living/simple_animal/hostile/carp/cayenne/lia
-	name = "Lia"
-	real_name = "Lia"
-	desc = "A failed experiment of Nanotrasen to create weaponised carp technology. This less than intimidating carp now serves as the Head of Security's pet."
-	faction = list("neutral")
-	health = 200
-	icon_dead = "magicarp_dead"
-	icon_gib = "magicarp_gib"
-	icon_living = "magicarp"
-	icon_state = "magicarp"
-	maxHealth = 200
-	random_color = FALSE
+/mob/living/simple_animal/hostile/carp/cayenne/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_DISK_VERIFIER, INNATE_TRAIT) //carp can verify disky
+	ADD_TRAIT(src, TRAIT_ADVANCEDTOOLUSER, INNATE_TRAIT) //carp SMART
+
+/mob/living/simple_animal/hostile/carp/cayenne/death(gibbed)
+	disky.forceMove(drop_location())
+	disky = null
+	return ..()
+
+/mob/living/simple_animal/hostile/carp/cayenne/examine(mob/user)
+	. = ..()
+	if(disky)
+		. += "<span class='notice'>Wait... is that [disky] in [p_their()] mouth?</span>"
+
+/mob/living/simple_animal/hostile/carp/cayenne/AttackingTarget(atom/attacked_target)
+	if(istype(attacked_target, /obj/item/disk/nuclear))
+		var/obj/item/disk/nuclear/potential_disky = attacked_target
+		if(potential_disky.anchored)
+			return
+		potential_disky.forceMove(src)
+		disky = potential_disky
+		to_chat(src, "<span class='nicegreen'>YES!! You manage to pick up [disky]. (Click anywhere to place it back down.)</span>")
+		regenerate_icons()
+		if(!disky.fake)
+			client.give_award(/datum/award/achievement/misc/cayenne_disk, src)
+		return
+	if(disky)
+		if(isopenturf(attacked_target))
+			to_chat(src, "<span class='notice'>You place [disky] on [attacked_target]</span>")
+			disky.forceMove(attacked_target.drop_location())
+			disky = null
+			regenerate_icons()
+			return
```

Comment:
Move this return outside of the if/else, since they both end in it.

Reply:
```suggestion
```